assets/model_monitoring/components/src/model_monitor_metric_publisher/run.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Entry script for Model Monitor Azure Monitor Metric Publisher Component."""
+
+import argparse
+import os
+
+from pyspark.sql import Row
+from typing import List
+
+from azure_monitor_metric_publisher import extract_location_from_aml_endpoint_str, publish_metric
+from shared_utilities.datetime_utils import parse_datetime_from_string
+from shared_utilities.io_utils import read_mltable_in_spark
+from shared_utilities.log_utils import log_error
+
+
+def run():
+    """Publish Azure monitor metrics."""
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument("--signal_metrics", type=str)
+    arg_parser.add_argument("--monitor_name", type=str)
+    arg_parser.add_argument("--signal_name", type=str)
+    arg_parser.add_argument("--data_window_start", type=str)
+    arg_parser.add_argument("--data_window_end", type=str)
+    args = arg_parser.parse_args()
+
+    metrics: List[Row] = read_mltable_in_spark(args.signal_metrics).collect()
+
+    try:
+        azureml_service_endpoint = os.environ['AZUREML_SERVICE_ENDPOINT']
```

Comment:
Who sets these envs? Also, it is possible to not have an online endpoint associated with the model monitor no? Then do we expect the env var to be an empty string or something like that?

Reply:
I think this is set by Execution Service (or AEther), which is also used for syncing AML run metadata etc.
This is not the "online endpoint" resource, but the AML service (cluster) endpoint, which usually looks like "https://eastus2.api.azureml.ms". I believe this is quite stable and unlikely to change. Of course, we can revisit if we see a problem later.